publicschema.json

```diff
@@ -646,8 +646,79 @@
                                         "on_success",
                                         "on_fail"
                                     ]
+                                },
+                                "max_auto_reruns": {
+                                    "description": "Maximum number of automatic reruns for this step",
+                                    "type": "integer",
+                                    "minimum": 1,
+                                    "maximum": 5
+                                },
+                                "auto_rerun_delay": {
+                                    "description": "Delay before automatic rerun of this step",
+                                    "type": "string",
+                                    "pattern": "^(10|[1-9])m$"
                                 }
-                            }
+                            },
+                            "allOf": [
+                                {
+                                    "if": {
+                                        "properties": {
+                                            "auto_rerun_delay": true
+                                        },
+                                        "required": ["auto_rerun_delay"]
+                                    },
+                                    "then": {
+                                        "required": ["max_auto_reruns"]
+                                    }
+                                },
+                                {
+                                    "if": {
+                                        "properties": {
+                                            "background": {
+                                                "const": true
+                                            }
+                                        },
+                                        "required": ["background"]
+                                    },
+                                    "then": {
+                                        "not": {
+                                            "anyOf": [
+                                                {
+                                                    "required": ["max_auto_reruns"]
+                                                },
+                                                {
+                                                    "required": ["auto_rerun_delay"]
+                                                }
+                                            ]
+                                        }
+                                    }
+                                },
+                                {
+                                    "if": {
```

Comment:
~~ is this schema rule not the same as the one above it? i.e. validates that `background: true` cannot be used with `max_auto_reruns` or `auto_rerun_delay`